bin/propolis-server/src/lib/stats.rs

```diff
@@ -156,13 +154,27 @@ pub async fn start_oximeter_server(
                 return Ok(server);
             }
             Err(e) => {
-                if retry_print_timeout == 0 {
-                    error!(log, "Can't connect to oximeter server:\n{}", e);
-                    retry_print_timeout = 1;
+                if attempts >= N_RETRY {
+                    error!(
+                        log,
+                        "Could not connect to oximeter after {} retries",
+                        N_RETRY
+                    );
+                    return Err(e.into());
                 }
-                // Retry every 10 seconds, but only print once a minute
-                retry_print_timeout = (retry_print_timeout + 1) % 7;
-                tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
+                attempts += 1;
+
+                warn!(
+                    log,
+                    "Could not connect to oximeter (retrying in {}s):\n{}",
+                    e,
+                    RETRY_WAIT_SEC
```

Comment:
These look like they should be in the other order.

Reply:
Derp, I totally missed this because the log messages are hard to read (#512).

Reply:
fixed in b9095d4